soc/nordic/nrf54l/soc.c

```diff
@@ -153,6 +153,11 @@ static int nordicsemi_nrf54l_init(void)
 	}
 
 #if (DT_PROP(DT_NODELABEL(vregmain), regulator_initial_mode) == NRF5X_REG_MODE_DCDC)
+#if defined(__CORTEX_M) && !defined(NRF_TRUSTZONE_NONSECURE) && defined(__ARM_FEATURE_CMSE)
```

Comment:
question: is vregmain available from riscv core? if not the cortex_m bit may be redundant.